client-js/app/elements/labrad-plot.ts

```diff
@@ -134,13 +143,90 @@ export class Plot extends polymer.Base {
 
 
   /**
-   * Redraw the plot and attach resize to the window resize event.
-   * Fires when the component is attached to the DOM.
+   * The base size of a point of data when in `dots` viewing mode.
    */
-  attached() {
-    this.redraw();
-    window.addEventListener('resize', (event) => this.redraw());
-  }
+  private dotBaseSize: number = 4;
+
+
+  /**
+   * A matrix for the use in transforming geometries.
+   */
+  private transformMatrix = new THREE.Matrix4();
+
+
+  /**
+   * A non-indexed plane unit geometry one world unit in size.
+   * Used to render rectangular data points.
+   */
+  private planeUnitGeometry =
+      new THREE.PlaneBufferGeometry(1, 1).toNonIndexed();
+
+
+  /**
+   * A short-hand to access the positions array in a unit plane.
+   */
+  private planeVertexPositions =
+      this.planeUnitGeometry.attributes.position.array;
+
+
+  /**
+   * The number of vertices in a plane geometry.
+   */
+  private planeVertexCount = this.planeVertexPositions.length / 3;
+
+
+  /**
+   * The default options for the plotting camera.
+   */
+  private cameraOpts = {
+    /** The field of view of the camera. */
+    fov: 45,
+
+    /** The initial aspect ratio of the camera. */
+    aspect: 1,
+
+    /** The near plane of the camera. */
+    near: -500,
+
+    /** The far plane of the camera. */
+    far: 500,
+
+    /** The starting position of the camera. */
+    startingPosition: new THREE.Vector3(0, 0, 1000)
+  };
+
+
+  /**
+   * A perspective camera to view the graph.
+   *
+   * The camera is currently only used for the sake of being able to observe
+   * the scene. Zooming/Panning is handled by `d3` rather than physically
+   * moving the camera around the 3D space provided by `THREE.js`. As such,
+   * camera properties such as position and zoom never change.
+   */
+  private camera = new THREE.PerspectiveCamera(
+    this.cameraOpts.fov,
+    this.cameraOpts.aspect,
+    this.cameraOpts.near,
+    this.cameraOpts.far);
+
+  /** The scene to render graphs to. */
```

Comment:
A lot of the comments here seem to just narrate the code and can probably be removed.